File: cacao/ui/components/data/badge/badge.js
(component) => {
    console.log("[CacaoCore] Rendering badge component:", component);
    const wrapper = document.createElement("span");
    wrapper.className = "badge-wrapper";

    // Render the main content if provided
    if (component.props.children) {
        wrapper.appendChild(window.CacaoCore.renderComponent(component.props.children));
    }

    // Create the badge element
    const badge = document.createElement("span");
    badge.className = "badge";

    // Handle dot style badge
    if (component.props.dot) {
        badge.classList.add("dot");
    } else if (component.props.count !== undefined) {
        // Show count if not zero or showZero is true
        if (component.props.count > 0 || component.props.showZero) {
            badge.textContent = component.props.count;
        }
    }

    wrapper.appendChild(badge);
    return wrapper;
}

File: cacao/ui/components/data/badge/badge.py
"""
Badge component for adding small indicators to elements.
"""

from typing import Any, Dict
from ...base import Component


class Badge(Component):
    """
    Adds a small indicator (e.g., a number or dot) to an element.
    """
    def __init__(
        self,
        count = None,
        children = None,
        dot = False,
        show_zero = False,
        **kwargs
    ) -> None:
        self.count = count
        self.children = children
        self.dot = dot
        self.show_zero = show_zero
        self.extra_props = kwargs

    def render(self) -> Dict[str, Any]:
        return {
            "type": "badge",
            "props": {
                "count": self.count,
                "children": self.children.render() if self.children else None,
                "dot": self.dot,
                "showZero": self.show_zero,
                **self.extra_props
            }
        }